src/main/java/com/exactpro/th2/crawler/processor/healer/grpc/HealerImpl.java

```diff
@@ -183,27 +180,30 @@ private void heal(Collection<EventData> events, Set<String> notFoundParent) thro
             if (cache.containsKey(parentId)) {
                 innerEvent = cache.get(parentId);
             } else {
-                if (notFoundParent.contains(parentId)) {
-                    LOGGER.info("Parent element {} none", parentId);
-                    return eventAncestors;
-                }
-
                 StoredTestEventWrapper parent = storage.getTestEvent(new StoredTestEventId(parentId));
-
                 if (parent == null) {
-                    Instant from = Instant.now();
-                    Instant to = from.plus(configuration.getWaitParent(), configuration.getWaitParentOffsetUnit());
-                    long sleepTime = Duration.between(from, to).abs().toMillis();
-                    LOGGER.info("Waiting for parentEventId in the interval from {} to {}. Wait for {}", from, to, Duration.ofMillis(sleepTime));
+                    if (notFoundParent.contains(parentId)) {
+                        LOGGER.info("Parent element {} none", parentId);
+                        return eventAncestors;
+                    }
+
+                    long from = System.currentTimeMillis();
+                    long to = from + Duration.of(configuration.getWaitParent(), configuration.getWaitParentOffsetUnit()).toMillis();
+                    long sleepTime = Duration.of(configuration.getWaitingStep(), configuration.getWaitingStepOffsetUnit()).toMillis();
+                    LOGGER.info("Waiting for parentEventId in an interval of size {} with a step {}.", Duration.ofMillis(to - from), Duration.ofMillis(sleepTime));
```

Comment:
Use debug level and check isDebugEnabled before logging, in because `Duration.ofMillis(sleepTime)` will be executed independently of configured log level.